### PR TITLE
docs(tree): add docs to cdk-tree

### PR DESCRIPTION
**Pull request by tinayuangao**

### CODE REVIEW

File: src/cdk/tree/tree.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+   </cdk-nested-tree-node>
+</cdk-tree>
+```
+
+Nested trees are easier to work with when hierarchical relationships are visually represented in
+ways that would be difficult to accomplish with flat nodes.
+
+### Using the CDK tree
+
+#### Wring your tree template
```
Comment (julianobrasil) [inline src/cdk/tree/tree.md]: typo: Wiring?

File: src/cdk/tree/tree.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+The `<cdk-tree>` enable developers to build customized tree for structured data.  The cdk-tree
+provides a foundation to build other features such as filtering on top of tree.
+For a Material design styled tree, see `<mat-tree>` which builds on top of the `<cdk-tree>`.
+
+There are two types of trees: Flat tree and Nested Tree. The DOM structures are different for
+these two types of trees.
+
+#### Flat tree
+
+In a flat tree, the hierarchy is flattened; nodes are not rendered inside of each other, but instead
```
Comment (jelbourn) [inline src/cdk/tree/tree.md]: There's some duplication between the cdk docs and the mat docs. Would it make sense to distinguish these similar to the table docs?

File: src/cdk/tree/tree.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+<cdk-tree-node *cdkNodeDef=“let node; when: isSpecial” cdkTreeNodePadding>
+    [ A special node {{node.value}} ]
+</cdk-tree-node>
+```
+
+### Data Source
+
+#### Connecting the tree to a data source
+
+Similar to `cdl-table`, data is provided to the tree through a `DataSource`. When the tree receives
```
Comment (jelbourn) [inline src/cdk/tree/tree.md]: "cdl" -> "cdk"

File: src/cdk/tree/tree.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+of data. Whenever the data source emits data to this stream, the tree will render an update.
+
+Because the data source provides this stream, it bears the responsibility of triggering tree
+updates. This can be based on anything: tree node expansion change, websocket connections, user
+interaction, model updates, time-based intervals, etc.
+
+
+#### Flat tree
+
+The flat tree data source should be responsible for the node expansion/collapsing events, since when
```
Comment (jelbourn) [inline src/cdk/tree/tree.md]: "should be responsible" -> "is responsible"

**Comment by tinayuangao** [comment]
Comments addressed. Please take a look. Thanks!

**Pull request merged**